digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/AwsService.scala

```diff
@@ -0,0 +1,299 @@
+import org.json4s._
+import org.json4s.jackson.Serialization
+import org.json4s.jackson.Serialization.write
+import fi.liikennevirasto.digiroad2.client.vvh.VVHClient
+import fi.liikennevirasto.digiroad2.oracle.OracleDatabase
+
+case class Dataset(datasetId: String, geoJson: Map[String, Any], roadlinks: List[List[BigInt]])
+
+class AwsService(vvhClient: VVHClient,
+                 val onOffLinearAssetService: OnOffLinearAssetService,
+                 val roadLinkService: RoadLinkService,
+                 val linearAssetService: LinearAssetService,
+                 val speedLimitService: SpeedLimitService,
+                 val pavedRoadService: PavedRoadService,
+                 val roadWidthService: RoadWidthService,
+                 val manoeuvreService: ManoeuvreService,
+                 val assetService: AssetService,
+                 val obstacleService: ObstacleService,
+                 val pedestrianCrossingService: PedestrianCrossingService,
+                 val railwayCrossingService: RailwayCrossingService,
+                 val trafficLightService: TrafficLightService,
+                 val massTransitLaneService: MassTransitLaneService,
+                 val numberOfLanesService: NumberOfLanesService
+                ) {
+
+  def withDynTransaction[T](f: => T): T = OracleDatabase.withDynTransaction(f)
+  def withDynSession[T](f: => T): T = OracleDatabase.withDynSession(f)
+  def awsDao: AwsDao = new AwsDao
+
+  implicit val formats = Serialization.formats(NoTypeHints)
+
+  final val AwsUser = "AwsUpdater"
+
+  val allDatasetStatus = Map(
+    0 -> "Inserted successfuly",
+    1 -> "Amount of features and roadlinks do not match",
+    2 -> "Errors in the features",
+    3 -> "Processed successfuly",
+    4 -> "Errors while processing"
+  )
+
+  val allFeatureStatus = Map(
+    0 -> "Inserted successfuly",
+    1 -> "Errors while validating",
+    2 -> "Processed successfuly",
+    3 -> "SpeedLimit with invalid speed",
+    4 -> "PavementClass with invalid pavement class",
+    5 -> "Obstacle with invalid class",
+    6 -> "Invalid sideCode"
+  )
+
+  private def linkIdValidation(linkIds: Set[Long]): Unit = {
+    val roadLinks = roadLinkService.getRoadsLinksFromVVH(linkIds)
+    linkIds.foreach { linkId =>
+      roadLinks.find(road => road.linkId == linkId && road.administrativeClass != State).get
+    }
+  }
+
+  private def validatePoint(datasetId: String, featureId: Long, properties: Map[String, Any], assetType: String): Unit = {
+    assetType match {
+      case "obstacle" =>
+        try {
+          val obstacleType = properties.getOrElse("class", 0).asInstanceOf[BigInt]
+          if (!Set(1, 2).contains(obstacleType.intValue())) {
+            awsDao.updateFeatureStatus(featureId, "5")
+            awsDao.updateDatasetStatus(datasetId, 2)
+          }
+        } catch {
+          case _ =>
+            awsDao.updateFeatureStatus(featureId, "5")
+            awsDao.updateDatasetStatus(datasetId, 2)
```

Comment:
This code is repeated too many times in the service

create a method for this updates where you can pass the error codes